cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -0,0 +1,144 @@
+        output["trustworthiness_score"] if output is not None else None for output in outputs
+    ]
+    df[f"{metadata_column_name}"] = [
+        output["response"] if output is not None else None for output in outputs
+    ]
+
+    if (
+        regex is None and return_values is None
+    ):  # we do not need to have a "log" column as original output is not augmented by regex or return values
+        return df[[f"{metadata_column_name}", f"{column_name_prefix}trustworthiness"]]
+
+    df[f"{column_name_prefix}log"] = [
+        output["response"] if output is not None else None for output in outputs
+    ]
+
+    if regex:
+        regex_list = get_compiled_regex_list(regex)
+        df[f"{metadata_column_name}"] = df[f"{metadata_column_name}"].apply(
+            lambda x: get_regex_match(x, regex_list, disable_warnings)
+        )
+
+    if return_values:
+        return_values_pattern = r"(" + "|".join(return_values) + ")"
+        df[f"{metadata_column_name}"] = df[f"{metadata_column_name}"].apply(
+            lambda x: get_return_values_match(
+                x, return_values, return_values_pattern, disable_warnings
+            )
+        )
+
+    return df[
+        [
+            f"{metadata_column_name}",
+            f"{column_name_prefix}trustworthiness",
+            f"{column_name_prefix}log",
+        ]
+    ]
+
+
+def get_regex_matches(
+    column_data: Union[pd.Series, List[str]],
+    regex: Union[str, re.Pattern[str], List[re.Pattern[str]]],
+    disable_warnings: bool = False,
+) -> Union[pd.Series, List[str]]:
+    """
+    Extracts the first match to the provided regular expression pattern from each example in the provided column of data.
+
+    Use this function for: tuning regex patterns to extract the best outputs from the raw LLM responses for your dataset obtained via ``enrich_data()``, without having to re-run the LLM.
+    If a list of regular expressions is provided, the expressions are applied in order, and the first valid regex match is returned.
+
+    **Note:** Regex patterns should each specify exactly 1 group that is represents the desired characters to be extracted from the raw response using parenthesis like so '(<desired match group pattern>)'.
+    **Example 1:** `r'.*The answer is: (Bird|[Rr]abbit).*'` will extract strings that are the words 'Bird', 'Rabbit' or 'rabbit' after the characters "The answer is: " from the raw response text.
+    **Example 2:** `r'.*(\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b).*'` will match an email in the raw response LLM response.
+
+    Args:
+        column_data: A pandas series or list of strings that you want to apply the regex to.
+        regex: A single string expression that will be passed into ``re.compile()``, a single compiled expression or a list of already compiled regular expressions.
+        disable_warnings: When True, warnings are disabled.
+
+    Returns:
+        The first matches of each response using the provided regex patterns.
```

Comment:
```suggestion
        column_data: A pandas Series or list of strings, where you want to apply a regex to extract matches from each element. This could be the `metadata` column output by ``enrich_data()``.
        regex: A string expression that will be passed into ``re.compile()``, a compiled regular expression, or a list of multiple already compiled regular expressions.
        disable_warnings: When True, warnings are disabled.

    Returns:
        Extracted matches to the provided regular expression from each element of the data column (specifically, the first match is returned).
```